av1an-core/src/settings.rs

```diff
@@ -628,11 +628,61 @@ properly into a mkv file. Specify mkvmerge as the concatenation method by settin
 
     if !self.force {
       self.validate_encoder_params();
+      self.check_rate_control();
     }
 
     Ok(())
   }
 
+  /// Checks if video parameters have rate control
+  /// and cq-level or target-bitrate set
+  /// Prints the warning
+  fn check_rate_control(&self) {
+    if self.encoder == Encoder::aom {
+      if !self.video_params.iter().any(|f| Self::check_end_usage(f)) {
+        println!(
+          "{}",
+          Color::Yellow
+            .paint("[WARN] Rate control mode was not provided, make sure to set `--end-usage=X`")
+        );
+      }
+
+      if !self.video_params.iter().any(|f| Self::check_rate(f)) {
+        println!("{}", Color::Yellow.paint("[WARN] Rate control value was not provided, make sure to set `--cq-level=X` or `--target-bitrate=X`"));
+      }
+    }
+  }
+
+  fn check_end_usage(s: &str) -> bool {
+    const END_USAGE: &str = "--end-usage=";
+    if s.len() <= END_USAGE.len() || !s.starts_with(END_USAGE) {
+      return false;
+    }
+
+    s.as_bytes()[END_USAGE.len()..]
+      .iter()
+      .all(|&b| (b as char).is_ascii_alphabetic())
+  }
+
+  fn check_rate(s: &str) -> bool {
```

Comment:
It feels like this function and `check_end_usage` should be renamed to reflect that it only applies to aomenc.